src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -176,36 +176,43 @@ public NadelExecutionStrategy(List<Service> services,
 
-                CompletableFuture<RootExecutionResultNode> convertedResult = serviceCallResult
-                        .thenApply(resultNode -> {
-                            if (nadelContext.getUserSuppliedContext() instanceof BenchmarkContext) {
-                                BenchmarkContext benchmarkContext = (BenchmarkContext) nadelContext.getUserSuppliedContext();
-                                benchmarkContext.serviceResultNodesToOverallResult.executionId = newExecutionContext.getExecutionId();
-                                benchmarkContext.serviceResultNodesToOverallResult.resultNode = resultNode;
-                                benchmarkContext.serviceResultNodesToOverallResult.overallSchema = overallSchema;
-                                benchmarkContext.serviceResultNodesToOverallResult.correctRootNode = resultNode;
-                                benchmarkContext.serviceResultNodesToOverallResult.fieldIdToTransformation = fieldIdToTransformation;
-                                benchmarkContext.serviceResultNodesToOverallResult.typeRenameMappings = typeRenameMappings;
-                                benchmarkContext.serviceResultNodesToOverallResult.nadelContext = nadelContext;
-                                benchmarkContext.serviceResultNodesToOverallResult.transformationMetadata = queryTransform.getRemovedFieldMap();
-                            }
-                            return (RootExecutionResultNode) serviceResultNodesToOverallResult
-                                    .convert(newExecutionContext.getExecutionId(),
-                                            resultNode,
-                                            overallSchema,
-                                            resultNode,
-                                            fieldIdToTransformation,
-                                            typeRenameMappings,
-                                            nadelContext,
-                                            queryTransform.getRemovedFieldMap());
-                        });
-
-                //set the result node count for this service
-                convertedResult.thenAccept(rootExecutionResultNode -> {
-                    resultComplexityAggregator.incrementServiceNodeCount(service.getName(), rootExecutionResultNode.getTotalNodeCount());
-                    resultComplexityAggregator.incrementFieldRenameCount(rootExecutionResultNode.getTotalFieldRenameCount());
-                    resultComplexityAggregator.incrementTypeRenameCount(rootExecutionResultNode.getTotalTypeRenameCount());
-                });
+                CompletableFuture<RootExecutionResultNode> convertedResult;
+
+                if (fieldIdToTransformation.size() > 0 || typeRenameMappings.size() > 0 || queryTransform.getRemovedFieldMap().hasRemovedFields() || queryTransform.getHintTypenameMap().size() > 0) {
```

Comment:
Whenever you have an if statement with this much logic, split it into a simple method.  So

```
if (fieldIdToTransformation.size() > 0 || typeRenameMappings.size() > 0 || queryTransform.getRemovedFieldMap().hasRemovedFields() || queryTransform.getHintTypenameMap().size() > 0) {
```
ends up looking like

```
if (doesNotNeedProcessing(arg1,arg2,arg3)) {
``

As a reader of code this is so much easier